rasa/src/Rasa/Internal/Listeners.hs

```diff
@@ -45,44 +45,44 @@ onEveryTrigger_ = void . onEveryTrigger
 -- Though arbitrary actions may be performed in the configuration block;
 -- it's recommended to embed such actions in the onInit event listener
 -- so that all event listeners are registered before anything Actions occur.
-onInit :: forall a. Action a -> Action ()
+onInit :: forall a. (Monoid a, Typeable a) => Action a -> Action ()
```

Comment:
I'm not sure what you have in mind for this monoid stuff, but can the monoidal result actually be used here? If not, I think the type should not change, and we could register `(const $ void action)` instead. This might also apply to the other void event registers.